x/evm/artela/contract/contract_func.go

```diff
@@ -178,7 +177,7 @@ func (k *AspectNativeContract) aspectsOf(ctx sdk.Context, method *abi.Method, co
 
 	aspectInfo := make([]types.AspectInfo, 0)
 	if isContract {
-		aspects, err := k.aspectService.GetAspectForAddr(ctx.BlockHeight()-1, contract)
+		aspects, err := k.aspectService.GetBoundAspectForAddr(ctx.BlockHeight()-1, contract)
```

Comment:
This could be wrong, what if this is a call? If this is a call, you are basically looking for `canonical block - 1`, which is not the latest one.